File: cpp/153_寻找旋转排序数组中的最小值_medium.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int n = nums.size();
        int l = 0, r = n-1;
        while (l <= r) {
            int mid = (l + r) / 2;
            if (nums[l] <= nums[mid])
                if (nums[mid] <= nums[r])
                    return nums[l];
                else
                    l = mid + 1;
            else
                r = mid;
        }   
        return nums[l];
    }
};

File: python/153_寻找旋转排序数组中的最小值_medium.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n = len(nums)
        l, r = 0, n-1
        while l <= r:
            mid = (l+r)//2
            if nums[l] <= nums[mid]:
                if nums[mid] <= nums[r]:
                    return nums[l]
                else:
                    l = mid + 1
            else:
                r = mid